cfgov/jinja2/v1/_includes/organisms/ask-search.html

```diff
@@ -54,7 +57,20 @@
                                name="q"
                                id="o-search-bar_query"
                                value="{{ ask_query }}"
-                               placeholder="{{ placeholder }}">
+                               placeholder="{{ placeholder }}"
+                               maxlength="{{ max_length }}"
+                               aria-describedby="o-search-bar-error_message">
+                    </div>
+                    <div class="a-form-alert a-form-alert__error u-hidden"
+                         id="o-search-bar-error_message"
+                         role="alert">
+                        {{ svg_icon('error-round') }}
+                        <p class="a-form-alert_text">
+                            Searches are limited to 50 characters.
```

Comment:
We could avoid a hard-coded value here.

```suggestion
                            Searches are limited to {{ autocomplete_max_chars }} characters.
```

Reply:
Nit to be ignored at will:   
I had noticed that freezing the input at 50 characters in practice limits a search value to 49 characters. I thought it might be easy to change the [test](https://github.com/cfpb/consumerfinance.gov/pull/6537/files#diff-80d79644e1dbe5feb3a4ca46b46b6815c7a1e6f522aace7b712d67de99587019R138) to `>` instead of `>=`, but that messes up the interactions, and it doesn't seem worth it to chase the semantic difference between whether the limit is the barrier or the last allowed value. It does the job well either way.

Reply:
Good call on using the new variable in the error message—I made that change in 8fb3d3c.

I went back and forth on whether the 50th or 51st character should trigger the error when I was setting this up. From the search data I've seen I don't think an extra character will make much of a difference, so I'm inclined to leave it as is for now.